core/modules/cart/helpers/productsEquals.ts

```diff
@@ -15,8 +15,12 @@ const getProductType = (product: CartItem): string =>
 const getServerItemId = (product: CartItem): string | number =>
   product.server_item_id || product.item_id
 
-const isServerIdsEquals = (product1: CartItem, product2: CartItem): boolean =>
-  getServerItemId(product1) === getServerItemId(product2)
+const isServerIdsEquals = (product1: CartItem, product2: CartItem): boolean => {
+  if (getServerItemId(product1) === undefined || getServerItemId(product2) === undefined) {
+    return false
```

Comment:
What if both products have empty server_item_id (say both products added in the offline mode and not yet synced); it’s probably from the caller context that it should be false
In this case correct?

Reply:
I should always be false when one of the products has no server_item_id or item_id other wise when both don't have it they are the same product but aren't in most cases.

Reply:
But for that case we have the check of the checksum.